infra/Pos/Subscription/Common.hs

```diff
@@ -88,14 +94,18 @@ subscriptionListener
     -> (ListenerSpec m, OutSpecs)
 subscriptionListener oq nodeType = listenerConv @Void oq $ \__ourVerInfo nodeId conv -> do
     mbMsg <- recvLimited conv
-    whenJust mbMsg $ \MsgSubscribe -> do
-      let peers = simplePeers [(nodeType, nodeId)]
-      bracket
-        (updatePeersBucket BucketSubscriptionListener (<> peers))
-        (\added -> when added $
-          void $ updatePeersBucket BucketSubscriptionListener (removePeer nodeId))
-        (\added -> when added $
-          void $ recvLimited conv) -- if not added, close the conversation
+    whenJust mbMsg $ \case
+        MsgSubscribe -> do
+            let peers = simplePeers [(nodeType, nodeId)]
+            bracket
+              (updatePeersBucket BucketSubscriptionListener (<> peers))
+              (\added -> when added $
+                void $ updatePeersBucket BucketSubscriptionListener (removePeer nodeId))
+              (\added -> when added $
+                void $ recvLimited conv) -- if not added, close the conversation
+        MsgSubscribeKeepAlive -> logDebug $ sformat
```

Comment:
This is not quite right. The `MsgSubscribeKeepAlive` at the top level here is for opening a conversation with that message, which in this case would always be a protocol error. The case where we expect to receive a keep-alive is in the `void $ recvLimited conv` line above. Previously that was doing a `recvLimited` in the expectation that no message would ever be received, and thereby keeping the connection open, now we need a loop that received and discards keep alive messages. Something like:

```
 -          void $ recvLimited conv) -- if not added, close the conversation
 +          forever $ do
 +            Just MsgSubscribeKeepAlive <- recvLimited conv
 +            logDebug ...
```
Ok, `forever` and the do style pattern matching is terrible error handling, but you get the idea.